src/storage/mapping/ExtensionBasedMapper.ts

```diff
@@ -64,10 +64,12 @@ export class ExtensionBasedMapper extends BaseFileIdentifierMapper {
     // If the extension of the identifier matches a different content-type than the one that is given,
```

Comment:
This comment is about the code block above this code but I can't select it since it wasn't changed in this PR.

I was first confused about why this code works, since this block is supposed to trigger if the target file exists, and if it does it will always return a content-type since `this.getContentTypeFromPath` will call `super.getContentTypeFromPath`, which will return octet stream as content-type. But the reason this still works is since the code assumes that a file only exists if `contentType` is defined. At least I assume that is the case, or maybe something else is happening that I missed so please investigate exactly what is going on there.

So at the very least I think we should move some of the changes and/or change some comments so everything still works as expected. For example, the `filePath.endsWith(`.${this.unknownMediaTypeExtension}` check fits better in the `ExtensionBasedMapper` instead of the `BaseFileIdentifierMapper` since this is the class that determines content-type based on the file extension.

Reply:
Yeah, the code block you mention is not really straightforward and maybe should be rewritten taking into account the up-to-date context.

> For example, the filePath.endsWith(.${this.unknownMediaTypeExtension}check fits better in theExtensionBasedMapperinstead of theBaseFileIdentifierMapper` since this is the class that determines content-type based on the file extension.

Agreed, but one issue we're having, is that `mapUrlToDocumentPath` in `BaseFileIdentifierMapper` is sometimes called with contentType set to undefined for other cases which messes with the unknown media type signal we're trying to communicate by setting contentType to undefined in `ExtensionBasedMapper`. Maybe we should not call the super function there? Or alternatively, change some types/interfaces so we can signal the unsupported media type unambiguously? (I started out this way, but then traced back my steps, because this introduces a lot of breaking changes)

Reply:
Another field somewhere might indeed be required, specifically to indicate if the target file exists or not, or something else, because I do like `contentType` being undefined being the indicator that no type could be determined. But this might be too much of a refactor so I'm also ok with just keeping the change as-is for now. There is another issue open to fully refactor the `FileMapper` classes anyway (#634). I'll leave it up to you if you want to go deeper into this or not.

Reply:
In that case, I would like to try fixing #634 next and leave this fix as-is for now!